core/src/test/scala/org/apache/spark/sql/SparkDataTypeTestSuite.scala

```diff
@@ -29,9 +29,9 @@ class SparkDataTypeTestSuite extends BaseTiSparkTest {
   test("double type test") {
     compSparkWithTiDB(
       qSpark =
-        "select id_dt, tp_double, tp_float, tp_real from full_data_type_table order by tp_double desc nulls first limit 10",
+        "select id_dt, tp_double, tp_float, tp_real from full_data_type_table order by tp_double desc nulls first, id_dt asc limit 10",
```

Comment:
why need to change this SQL?
Is the Spark 3.4 change the order of returned data.

Reply:
That is because in the returned data, there are two rows where the tp_double columns are equal, resulting in some randomness in their ordering. Adding the "id_dt asc" clause eliminates this randomness when the tp_double columns have identical values.